File: node-server/src/models/coin.model.ts
import { Schema, Document, model } from 'mongoose'

export interface CoinDocument extends Document {
    _id: string;
    name: string;
    connection: string;
    image: string;
    lastPrice: number;
}

const coinSchema = new Schema({
   name: {
       type: String,
       required: true
   },
    connection: {
       type: String,
       required: true
    },
    image: {
       type: String,
       required: true
    },
    lastPrice: {
       type: Number,
       required: true,
       default: 0
    }
});

const Coin = model<CoinDocument>('Coin', coinSchema);

export default Coin

File: flutter_mobile/lib/models/coin.model.dart
import 'package:fluent_ui/fluent_ui.dart';
import 'package:flutter_mobile/core/url-builder.core.dart';

class Coin {
  static const String apiId = '_id';
  final String id;

  static const String apiName = 'name';
  final String name;

  static const String apiConnection = 'connection';
  final String connection;

  static const String apiLastPrice = 'lastPrice';
  final double price;

  static const String apiImage = 'image';
  final String _image;

  Coin({
    required this.id,
    required this.name,
    required this.connection,
    required this.price,
    required String image
  }) : _image = image;

  factory Coin.fromJson(Map<String, dynamic> json) {
    return Coin(
      id: json[apiId],
      name: json[apiName],
      connection: json[apiConnection],
      price: 1.0 * json[apiLastPrice],
      image: json[apiImage]
    );
  }

  String get image {
    return _image;
  }

}